packages/cycle-scripts-one-fits-all/template/src/drivers.ts

```diff
@@ -10,50 +10,55 @@ import storageDriver from '@cycle/storage';
 
 import speechDriver from './drivers/speech';
 
-let mkDriversCond: any;
-/// #if PRODUCTION
-mkDriversCond = () => ({
-    DOM: makeDOMDriver('#app'),
-    HTTP: makeHTTPDriver(),
-    time: timeDriver,
-    router: makeRouterDriver(
-        createBrowserHistory(),
-        switchPath as RouteMatcher
-    ),
-    storage: storageDriver,
-    speech: speechDriver
-});
-
-/// #else
-mkDriversCond = () => ({
-    DOM: restartable(makeDOMDriver('#app'), {
-        pauseSinksWhileReplaying: false
-    }),
-    HTTP: restartable(makeHTTPDriver()),
-    time: timeDriver,
-    router: makeRouterDriver(
-        createBrowserHistory(),
-        switchPath as RouteMatcher
-    ),
-    storage: storageDriver,
-    speech: speechDriver
-});
-/// #endif
-export const mkDrivers = mkDriversCond;
+export type DriverThunk = [string, () => any];
+export type DriverThunkMapper = ( t: DriverThunk) => DriverThunk;
+
+const driverThunks : DriverThunk[] = [
+    ['DOM', () => makeDOMDriver('#app')],
+    ['HTTP', () => makeHTTPDriver()],
+    ['time', () => timeDriver],
+    [
+        'router',
+        () =>
+            makeRouterDriver(createBrowserHistory(), switchPath as RouteMatcher)
+    ],
+    ['storage', () => storageDriver],
+    ['speech', () => speechDriver],
+    [
+        'auth0',
+        () =>
+            makeAuth0Driver(
+                'CoDxjf3YK5wB9y14G0Ee9oXlk03zFuUF',
+                'odbrian.eu.auth0.com'
+            )
+    ]
+];
+
+export const buildDrivers = (fn : DriverThunkMapper) =>
+    driverThunks
+        .map(fn)
+        .map(([n, t] : DriverThunk) => ({ [n]: t }))
+        .reduce((a, c) => Object.assign(a, c), {});
```

Comment:
nope. The callback gets 4 parameters which end up causing extra members. Took me a while to figure that.